### DISCUSSION TITLE
Transitioner question about disabling transitions

### DISCUSSION THREAD

**Discussion by jbooth88** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3120] [Q&A] [answered]
I just started playing with the Transitioner control in this toolkit.
I like how easy it is to use this control as a paged wizard, but I wondering if it's possible to disable all transition effects. 

I tried implementing techniques explained in the wiki but didn't get anywhere. 
Using TransitionerSlide and BackwardWipe sometimes had an effect, sometimes not.
TransitionAssist.DisableTransitions="True" had no effect whatsoever

Any suggestions?

**Comment by nicolaihenriksen** [comment, marked as answer]
@jbooth88 I have not played around with the `Transitioner` a whole lot, but I just spent 10 minutes playing with it in the demo tool.

I agree that `TransitionAssist.DisableTransitions="True"` seems to have no effect whatsoever. Surprised me a bit, but to be fair, the documentation hints that it won't work in all places; perhaps the `Transitioner` itself is one of the places it does not:
> Allows transitions to be disabled where supported.

However, I think you should be able to accomplish what you want by levering the `TransistionerSlide.ForwardWipe` and `TransistionerSlide.BackwardWipe`. If you try setting all of them on all your slides to a `SlideWipe` with zero duration I think you get closer to what you want. Something like this:

```xaml
<materialDesign:TransitionerSlide>
  <materialDesign:TransitionerSlide.BackwardWipe>
    <materialDesign:SlideWipe Duration="0" />
  </materialDesign:TransitionerSlide.BackwardWipe>
  <materialDesign:TransitionerSlide.ForwardWipe>
    <materialDesign:SlideWipe Duration="0" />
  </materialDesign:TransitionerSlide.ForwardWipe>
    <!-- slide content goes here -->
</materialDesign:TransitionerSlide>
```

The key thing to note here is that a transition from slide A to slide B is not necessarily the same transition used in the transition from slide B to slide A. Each slide defines a transition moving forward or backwards it seems; so to disable the transitions, you will need to disable both forward and backward for all slides. At least that is my take on it.

Again, I don't really have a lot of knowledge of this control, so I was just playing around with it.

**Reply by jbooth88** [reply]
Thanks for the help. I missed the ForwardWipe property. Your suggestion did get the effect I wanted. 

Instead however I tried setting these properties to null. I also set OpeningEffect to null in case it had a default animation. This appears to have the same effect. I assume this second approach has better performance, because maybe this way the framework will skip trying to draw a storyboard with zero duration? I'm not experienced with WPF animation framework so I honestly have no idea which approach is better.

I really like how easy it is to use this control to simulate paged navigation. :)

```
<materialDesign:TransitionerSlide ForwardWipe="{x:Null}" BackwardWipe="{x:Null}" OpeningEffect="{x:Null}">
    <!-- slide content goes here -->
</materialDesign:TransitionerSlide>
```

**Reply by nicolaihenriksen** [reply]
Perfect, I agree that using `{x:Null}` is probably a better approach! 👍
